### ISSUE TITLE
[v3] Dynamic (discoverable) providers

### ISSUE THREAD

**Issue by andersevenrud** [os-js/OS.js#707]
At the moment packages (apps, themes, icons, sounds) can be discovered and automatically used by client/server via `package:discover`.

It would be cool if this could be added to service providers as well.

Basically use the same approach, except change `package.json` to (instead of `package` value):

```
{
  "osjs": {
    "type": "service"
  }
}
```

Then have the CLI generate a list (just like `/packages.json` from packages discovery) which is used by the `index.js` bootstrap script.

A downside with this is that you'd have to explicitly set all options in `config.js` - but that's not a big deal.

Maybe look into configuration publishing as well to prevent that.

**Comment by 17lwinn** [comment]
Yep, would be better if i didn't have to build after installing providers

**Comment by andersevenrud** [comment]
@17lwinn You'd still have to rebuild after adding a client service provider.

**Comment by 17lwinn** [comment]
I know but other providers should not require a rebuild
